--- libqalculate-0.9.10/libqalculate.pc.in.orig	2016-11-28 17:36:05.568065567 -0500
+++ libqalculate-0.9.10/libqalculate.pc.in	2016-11-28 17:37:08.583745951 -0500
@@ -5,7 +5,7 @@
 
 Name: libqalculate
 Description: libqalculate
-Requires: glib-2.0 libxml-2.0
+Requires.private: glib-2.0 libxml-2.0
 Version: @VERSION@
 Libs: -L${libdir} -lqalculate
-Cflags: -I${includedir}
\ No newline at end of file
+Cflags: -I${includedir}
